packages/Manager/src/main/resources/db/migration/V0040__observed_water_use_aggregated_daily.sql

```diff
@@ -0,0 +1,277 @@
+DROP VIEW IF EXISTS observed_water_use_aggregated_daily CASCADE;
+
+CREATE OR REPLACE VIEW observed_water_use_aggregated_daily AS
+
+WITH filtered_obs AS (SELECT osm.id,
+                             osm.site_id,
+                             os.name as site_name,
+                             date(o.observed_at AT TIME ZONE 'NZST') AS day_observed_at,
+                             osm.measurement_name,
+                             o.amount
+                      FROM observations o
+                               LEFT JOIN observation_sites_measurements osm ON o.observation_measurement_id = osm.id
+                               LEFT JOIN observation_sites os ON os.id = osm.site_id
+                      WHERE osm.measurement_name IN ('Water Meter Reading', 'Water Meter Volume')
+                        AND o.observed_at > '2022-11-01 00:00:00+12'),
+
+daily_calculation AS (
+
+SELECT filtered_obs.site_name,
+       filtered_obs.day_observed_at,
+       filtered_obs.measurement_name,
+       CASE
+           -- first condition: caculate usage for water meter volume measurements
+           WHEN filtered_obs.measurement_name = 'Water Meter Volume' THEN SUM(filtered_obs.amount)
+           -- second condition: return null for observations less frequent than daily
+	         WHEN count(*) = 1 AND LAG(max(day_observed_at)) OVER (PARTITION BY id, measurement_name ORDER BY id, filtered_obs.day_observed_at, measurement_name) != day_observed_at - interval '1 day' THEN NULL
+           -- third condition: return usage by comparing with the record from the previous day for observations recorded once per day 
+	         WHEN count(*) = 1 THEN max(filtered_obs.amount) - LAG(max(filtered_obs.amount)) OVER (PARTITION BY id, measurement_name ORDER BY id, filtered_obs.day_observed_at, measurement_name)
+            -- fourth condition: return usage based on observations within the day
+           WHEN filtered_obs.measurement_name = 'Water Meter Reading' THEN MAX(filtered_obs.amount) - MIN(filtered_obs.amount)
+           END AS daily_usage
+FROM filtered_obs
+GROUP BY filtered_obs.id,
+         filtered_obs.site_name,
+         filtered_obs.day_observed_at,
+         filtered_obs.measurement_name
+),
+
+duplication_dates AS (
+    SELECT
+        site_name,
+        day_observed_at,
+        MIN(day_observed_at) OVER (PARTITION BY site_name) AS min_duplication_date
+    FROM
+        daily_calculation
+    GROUP BY
+        site_name, day_observed_at
+    HAVING
+        COUNT(DISTINCT measurement_name) > 1
+),
+
+PreviousMeasurement AS (
```

Comment:
This, and others ... keep the naming consistent i.e. `snake_case` which is fairly standard for Postgres.